spring-cloud-gcp-kms/src/main/java/com/google/cloud/spring/kms/KMSTemplate.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.kms;
+
+import java.util.Base64;
+
+import com.google.cloud.kms.v1.CryptoKeyName;
+import com.google.cloud.kms.v1.DecryptRequest;
+import com.google.cloud.kms.v1.DecryptResponse;
+import com.google.cloud.kms.v1.EncryptRequest;
+import com.google.cloud.kms.v1.EncryptResponse;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.spring.core.GcpProjectIdProvider;
+import com.google.protobuf.ByteString;
+
+/**
+ * Offers convenience methods for performing common operations on KMS including
+ * encrypting and decrypting text.
+ *
+ * @author Emmanouil Gkatziouras
+ */
+public class KMSTemplate implements KMSOperations {
+
+	private final KeyManagementServiceClient client;
+
+	private final GcpProjectIdProvider projectIdProvider;
+
+	public KMSTemplate(
+			KeyManagementServiceClient keyManagementServiceClient,
+			GcpProjectIdProvider projectIdProvider) {
+		this.client = keyManagementServiceClient;
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	public String encrypt(String cryptoKey, String plaintext) {
+		CryptoKeyName cryptoKeyName = KMSPropertyUtils.getCryptoKeyName(cryptoKey, projectIdProvider);
+
+		ByteString plaintextByteString = ByteString.copyFromUtf8(plaintext);
+
+		EncryptRequest request = EncryptRequest.newBuilder()
+				.setName(cryptoKeyName.toString())
+				.setPlaintext(plaintextByteString)
+				.build();
+
+		EncryptResponse response = client.encrypt(request);
```

Comment:
Should we add an option to perform integrity verification on response?
See: https://cloud.google.com/kms/docs/encrypt-decrypt

Reply:
Yes indeed. When I started I used the guava Crc32C hashing function `Hashing.crc32c` on the old repo. However due to being annotated as `@Beta` it was failing.

I will bring back the Crc32C functionality and let me know how to proceed.

Reply:
So I did put the changes with the crc32c verification and indeed there is a failure due to the `@Beta` annotation from Guava.
https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/175/checks?check_run_id=1552995546#step:5:760 

If there is a workaround for this will proceed as requested.

Reply:
This is just a checkstyle violation. We may just have to disable the rule on this file or something. When you build locally you can just add `-Dcheckstyle.skip` to your maven command.

Reply:
Perfect! I thought it was a strict rule. Let me know what is the preferable workaround.

For example can be disabled through maven
```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-checkstyle-plugin</artifactId>
				<executions>
					<execution>
						<id>checkstyle-validation</id>
						<phase>validate</phase>
						<goals>
							<goal>check</goal>
						</goals>
						<configuration>
							<!-- Checkstyle rules inherited from spring-cloud-build -->
							<suppressionsLocation>src/checkstyle/checkstyle-suppressions.xml</suppressionsLocation>
....
							<excludes>**/KmsTemplate.java</excludes>
						</configuration>
					</execution>
				</executions>
			</plugin>
```

Reply:
This works for now, but it will disable all rules on the file. I think you can also disable the specific rule on the specific file in `checkstyle-suppressions.xml`.

Reply:
Thanks. Moved it there!

```
<suppressions>
    <suppress files="[\\/]spring-cloud-gcp-samples[\\/]" checks="HideUtilityClassConstructorCheck" />
    <suppress files="package\-info\.java" checks="RegexpHeader" />
    <suppress files=".*" checks="LineLength" />
    <suppress checks="Illegal" files=".*RSAKeyGeneratorUtils.java"/>
    <suppress checks="IllegalImport" files=".*KmsTemplate.java"/>
</suppressions>
```

Reply:
Due to being a granular configuration I set this as resolved. Let me know otherwise.